ui/raidboss/data/03-hw/dungeon/gubal_library_hard.js

```diff
@@ -0,0 +1,300 @@
+'use strict';
+
+// The Great Gubal Library--Hard
+[{
+  zoneRegex: /Great Gubal Library \(Hard\)/,
+  timelineFile: 'gubal_library_hard.txt',
+  timelineTriggers: [
+    {
+      id: 'Gubal Hard Triclip',
+      regex: /Triclip/,
+      beforeSeconds: 5,
+      condition: function(data) {
+        return data.role == 'healer' || data.role == 'tank';
+      },
+      infoText: {
+        en: 'Tank Buster',
+      },
+    },
+    {
+      id: 'Gubal Hard Searing Wind',
+      regex: /Searing Wind/,
+      beforeSeconds: 5,
+      infoText: function(data) {
+        if (data.role == 'tank') {
+          return {
+            en: 'Tank Cleave on you',
+          };
+        }
+        return {
+          en: 'Avoid tank cleave',
+        };
+      },
+    },
+    {
+      id: 'Gubal Hard Properties of Darkness',
+      regex: /Darkness \(buster\)/,
+      beforeSeconds: 5,
+      condition: function(data) {
+        return data.role == 'healer' || data.role == 'tank';
+      },
+      infoText: {
+        en: 'Tank Buster',
+      },
+    },
+  ],
+  triggers: [
+    {
+      id: 'Gubal Hard Bibliocide',
+      regex: / 14:1945:Liquid Flame starts using Bibliocide/,
+      condition: function(data) {
+        return data.role == 'healer';
+      },
+      infoText: {
+        en: 'AoE',
+      },
+    },
+    {
+      // Ferrofluid handling is shamelessly copied from Hades Normal
+      id: 'Gubal Hard Ferrofluid Collect',
+      regex: / 1B:\y{ObjectId}:(\y{Name}):....:....:(0030|0031):/,
+      run: function(data, matches) {
+        data.fluid = data.fluid || {};
+        if (matches[2] == '0030')
+          data.fluid[matches[1]] = 'positive';
+        if (matches[2] == '0031')
+          data.fluid[matches[1]] = 'negative';
+      },
+    },
+    {
+      id: 'Gubal Hard Ferrofluid Execute',
+      regex: / 1B:\y{ObjectId}:\y{Name}:....:....:0030|0031:/,
```

Comment:
Should the or be within parentheses?

Reply:
Um. Wow. You're right, it should definitely be `(?:0030|0031)`. I'm a little surprised this didn't fail for me at least once, 25% potential failure rate isn't that low.